### PR TITLE
Add support to CRAM-MD5 login

### PR DESCRIPTION
**Pull request by manugc**

### CODE REVIEW

**Comment by dantti** [comment]
Why did you change from QTcpSocket to QSslSocket? You will break non encrypted traffic.

**Comment by manugc** [comment]
> Why did you change from QTcpSocket to QSslSocket? You will break non encrypted traffic.

Hola, no QSslSocket hereda de QTcpSocket, mientras que no comenzará la encriptación funciona igual. De esta manera evitamos los cast.

**Comment by manugc** [comment]
Hi, no QSslSocket inherits from QTcpSocket, while encryption will not start working the same. In this way we avoid cast.

**Comment by manugc** [comment]
Hi, can you tell me something about the proposed changes?

**Comment by manugc** [comment]
Hi, can we say that this repository is abandoned?

File: src/mimemessage.cpp [hunk 1]
```diff
@@ -68,11 +68,33 @@ bool MimeMessage::write(QIODevice *device)
     // Headers
     QByteArray data;
 
+    if(d->listExtraHeaders.count())
+    {
+        data.clear();
+        QListIterator<QByteArray> iExtraHeaders(d->listExtraHeaders);
+        while (iExtraHeaders.hasNext())
```
Comment (dantti) [inline src/mimemessage.cpp]: Please use Stl style iterators

File: src/mimemessage.cpp [hunk 2]
```diff
@@ -68,11 +68,33 @@ bool MimeMessage::write(QIODevice *device)
     // Headers
     QByteArray data;
 
+    if(d->listExtraHeaders.count())
+    {
+        data.clear();
```
Comment (dantti) [inline src/mimemessage.cpp]: There's no need to clear an empty data.

File: src/mimemessage_p.h [hunk 3]
```diff
@@ -40,6 +40,8 @@ class MimeMessagePrivate
     MimePart *content = nullptr;
     MimePart::Encoding encoding = MimePart::_8Bit;
     bool autoMimeContentCreated;
+    EmailAddress replyTo;
+    QList<QByteArray> listExtraHeaders;
```
Comment (dantti) [inline src/mimemessage_p.h]: replace with std::vector

**Comment by dantti** [comment]
An no it's not abandoned you can look at the history...

**Comment by dantti** [comment]
And coding style should usually put '{' on the same IF or WHILE lines.

**Comment by dantti** [comment]
I've cherry picked the ok commits, thanks

**Pull request closed without merging**